src/plotting.py

```diff
@@ -1,39 +1,70 @@
-                            np.degrees(angles[1]), 
-                            color=[0, 0.5, 0],  # Make angles green
-                            alpha=0.5)  # Make the wedges transparent
-    ax.add_patch(beam)
-    return beam
-
-def plot_beams(beam_list, cell_radius, ax):
-    beam_plots = []
-    for start_angle, end_angle, _ in beam_list:
-        beam = plot_beam((start_angle, end_angle), cell_radius, ax)
-        beam_plots.append(beam)
-    return beam_plots
-
-def plot_users(users: Union[User, Sequence[User]], ax: Axes):
-    user_plots = []
-    for user in users:
-        user_plot = ax.plot(user.x, user.y, marker='o', c="tab:blue")
-        user_plots.append(user_plot)
-    return user_plots
-
-def plot_user_path(users: Union[User, Sequence[User]], ax: Axes, num_steps=None):
-    if num_steps is not None:
-        num_steps = num_steps + 1
-    path_plots = []
-    for user in users:
-        path_plot = ax.plot(user.path[0, 0:num_steps], user.path[1, 0:num_steps])
-        path_plots.append(path_plot)
-        # ax.plot(user.path[0,0], user.path[1,0], 'go')
-        # ax.plot(user.path[0,num_steps], user.path[1,num_steps], 'ro')
-    return path_plots
+# Circular Cell
+class Cell:
+    def __init__(self, radius = 1): # Defined by radius
+        self.radius = radius
+        self.q = np.linspace(0,2*np.pi,50)
+        self.x = radius*np.cos(self.q)
+        self.y = radius*np.sin(self.q)
+        self.borders = np.array([0, np.pi/2, np.pi, 3*np.pi/2, 2*np.pi])
+
+# Plot Cell, Users, and Beams
+class Plot:
+    def __init__(self, cell: Cell): # Requires a Cell
+        self.cell = cell
+        self.fig, self.ax = plt.subplots()
+        self.ax.plot(cell.x, cell.y, color='tab:blue')
+        for angle in cell.borders:
+            if angle == 2*np.pi:
+                break
+            ri = 1
+            qi = angle
+            x = ri*np.cos(qi)
+            y = ri*np.sin(qi)
+            self.ax.plot([0, x], [0, y], color='tab:blue')
+
+        self.points = [] # Member defines users on plot
+
+    def add_beam(self, beam: Beam): # To plot a single Beam
+        wedge = mpatches.Wedge((0.,0.), beam.length,
+                                    np.degrees(beam.start_angle),
+                                    np.degrees(beam.end_angle),
+                                    color = beam.color, # color of beam
+                                    alpha = 0.5) # Make the wedges transparent
+        self.ax.add_patch(wedge)
+
+    def clear_beams(self):
+        [p.remove() for p in reversed(self.ax.patches)]
+
+    def add_beams(self, beams: Union[Beam, Sequence[Beam]]): # To plot multiple beams
+        for beam in beams:
+            self.add_beam(beam)
+
+    def add_users(self, users: Union[User, Sequence[User]]): # To plot multiple users
+        self.points = []
+        for user in users:
+            self.points.extend(
+                self.ax.plot(user.x, user.y, marker='o', color='tab:blue')
+            )
```

Comment:
We use reverse because we are modifying the list that we are iterating. More specifically, we remove the last item from the list each iteration. If we removed the first item each time we would experience ghosting due to the list structure also changing. i.e. we would not remove all of the beams.

Reply:
I think that that would be the case if we were doing `list.remove(list_member)`. But remove is a function in the Axes class (inherited from the Artist class). So we're doing `list_member.remove()`, which doesn't necessarily affect any list.

Reply:
See https://matplotlib.org/stable/api/_as_gen/matplotlib.artist.Artist.remove.html